pallets/gear/src/tests.rs

```diff
@@ -13164,3 +13164,60 @@ fn double_read_works() {
         );
     });
 }
+
+/// Tests gas allowance exceed handling.
+/// More precisely, it checks one property:
+/// no context data is stored within previously
+/// executed message when gas allowance exceed
+/// happened.
+#[test]
+fn test_gas_allowance_exceed() {
+    use crate::QueueProcessingOf;
+    use common::storage::{Counted, Queue, Toggler};
+
+    init_logger();
+    new_test_ext().execute_with(|| {
+        let wat = r#"
+            (module
+            (import "env" "memory" (memory 1))
+            (import "env" "gr_reply" (func $reply (param i32 i32 i32 i32)))
+            (export "handle" (func $handle))
+            (func $handle
+                (call $reply (i32.const 0) (i32.const 32) (i32.const 10) (i32.const 333))
```

Comment:
why reply

Reply:
It changes the context during execution. The aim was to test that it's not saved on gas allowance exceeded.

Reply:
> It changes the context during execution. The aim was to test that it's not saved on gas allowance exceeded.

Yeah, but even if reply wasn't sent context may be stored (for wait wakes case)
Please also expand test that after wait-wake in similar situation we have correct context